GitUI/UserControls/GitBlameParser.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Diagnostics;
+using System.Text.RegularExpressions;
+using GitCommands;
+using GitExtUtils;
+using GitUIPluginInterfaces;
+
+namespace GitUI.UserControls;
+
+internal partial class GitBlameParser : IGitBlameParser
+{
+    private readonly Func<IGitModule> _getModule;
+
+    [GeneratedRegex(@"\@\@ -(?<PreviousLineNumber>\d+)(,(?<LineRemovedCount>\d+))? \+(?<CurrentLineNumber>\d+)(,(?<LineAddedCount>\d+))? \@\@", RegexOptions.ExplicitCapture)]
+    private static partial Regex ChunkHeaderRegex();
+
+    public GitBlameParser(Func<IGitModule> getModule)
+    {
+        _getModule = getModule;
+    }
+
+    public int GetOriginalLineInPreviousCommit(GitRevision selectedBlamedRevision, string filename, int selectedLine)
+    {
+        ObjectId? parentId = selectedBlamedRevision!.FirstParentId;
+
+        if (parentId == null)
```

Comment:
Why have removed `selectedLine == 1 ||`?

Reply:
Because special case where selected line is the 1st one is perfectly handled by the `GetOriginalLineInPreviousCommit()`.
It was a little shortcut to avoid running some code for this specific use case but I reevaluate and think it does not worth it as I'm pretty sure no one will ever blame the first line 😉 (and if he does it, that will be ok).